### ISSUE TITLE
Computing objective for discrete skills

### ISSUE THREAD

**Issue by chengxuxin** [artberryx/LSD#3]
In lsd.py    

I don't really understand the difference between `target_z` and `logits`. I think they are the same vectors with the same values.

What is the purpose of projecting `target_z` to diagonal and then sum back?

I would really appreciate it if you could explain the difference here.

```
def _get_te_option_log_probs(self, tensors, v):
        # Note: actually te_option_log_probs isn't a log prob

        cur_z = self.traj_encoder(v['obs_flat'])[0].mean
        next_z = self.traj_encoder(v['next_obs_flat'])[0].mean
        target_z = next_z - cur_z

        if self.discrete:
            target_z = target_z.reshape(target_z.size(0), 1, target_z.size(1))
            eye_z = torch.eye(self.dim_option, device=target_z.device).reshape(1, self.dim_option, self.dim_option).expand(target_z.size(0), -1, -1)
            logits = (eye_z * target_z).sum(dim=2)
            masks = (v['options_flat'] - v['options_flat'].mean(dim=1, keepdim=True)) * (self.dim_option) / (self.dim_option - 1 if self.dim_option != 1 else 1)
            te_option_log_probs = (logits * masks).sum(dim=1)
        else:
            inner = (target_z * v['options_flat']).sum(dim=1)
            te_option_log_probs = inner
```

**Comment by seohongpark** [comment]
Hi Xuxin,

You're right, this is due to the legacy code before we cleaned it up. Indeed `target_z` and `logits` are the same.
